src/track/macro.h

```diff
@@ -0,0 +1,112 @@
+#pragma once
+
+#include <QtCore>
+#include <memory>
+
+#include "engine/engine.h"
+#include "proto/macro.pb.h"
+#include "util/db/dbid.h"
+namespace proto = mixxx::track::io;
+
+const QLoggingCategory macroLoggingCategory("macros");
+constexpr int kMacrosPerChannel = 16;
+
+/// A MacroAction is the building block of a Macro.
+/// It contains a position as well as the action to be taken at that position.
+///
+/// Note that currently only jumps to a target position are available, but that
+/// is subject to change.
+struct MacroAction {
+    enum class Type : uint32_t {
+        Jump = 0
+    };
+
+    // use FramePos once https://github.com/mixxxdj/mixxx/pull/2961 is merged
+    const double sourceFrame;
```

Comment:
const members prevent copying

Reply:
yep, but mutability is always a danger in OOP, so can we somehow eliminate the copying instead?

Reply:
@Holzhaus @uklotzde :)

Reply:
I think you need to implement a copy constructor.

Reply:
You should not need to implement a copy or move constructor manually, except for some very special cases. Declaring members as `const` doesn't work for value classes. The class could still be almost immutable if the only mutation is copy/move assignment. Remember, you could use always `const` in the outer context to make this instance immutable.

Reply:
If you make this an actual class with private members and getters, you can also just not implement any setters.